// ng2-chartjs2/components/chart.ngfactory.ts
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/render/api';
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/linker/element';
import * as import3 from 'ng2-chartjs2/components/chart';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import5 from '@angular/core/src/di/injector';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from '@angular/core/src/linker/element_ref';
import * as import9 from '@angular/core/src/metadata/view';
import * as import10 from '@angular/core/src/linker/component_factory';
var renderType_ChartComponent_Host:import0.RenderComponentType = (null as any);
class _View_ChartComponent_Host0 extends import1.AppView<any> {
  _el_0:any;
  /*private*/ _appEl_0:import2.AppElement;
  _ChartComponent_0_4:import3.ChartComponent;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement) {
    super(_View_ChartComponent_Host0,renderType_ChartComponent_Host,import6.ViewType.HOST,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import2.AppElement {
    this._el_0 = this.selectOrCreateHostElement('chart',rootSelector,(null as any));
    this._appEl_0 = new import2.AppElement(0,(null as any),this,this._el_0);
    var compView_0:any = viewFactory_ChartComponent0(this.viewUtils,this.injector(0),this._appEl_0);
    this._ChartComponent_0_4 = new import3.ChartComponent(new import8.ElementRef(this._el_0));
    this._appEl_0.initComponent(this._ChartComponent_0_4,[],compView_0);
    compView_0.create(this._ChartComponent_0_4,this.projectableNodes,(null as any));
    this.init([].concat([this._el_0]),[this._el_0],[],[]);
    return this._appEl_0;
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import3.ChartComponent) && (0 === requestNodeIndex))) { return this._ChartComponent_0_4; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    if (((this.numberOfChecks === 0) && !throwOnChange)) { this._ChartComponent_0_4.ngOnInit(); }
    if (!throwOnChange) { this._ChartComponent_0_4.ngDoCheck(); }
    this.detectContentChildrenChanges(throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
  destroyInternal():void {
    this._ChartComponent_0_4.ngOnDestroy();
  }
}
function viewFactory_ChartComponent_Host0(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement):import1.AppView<any> {
  if ((renderType_ChartComponent_Host === (null as any))) { (renderType_ChartComponent_Host = viewUtils.createRenderComponentType('',0,import9.ViewEncapsulation.None,[],{})); }
  return new _View_ChartComponent_Host0(viewUtils,parentInjector,declarationEl);
}
export const ChartComponentNgFactory:import10.ComponentFactory<import3.ChartComponent> = new import10.ComponentFactory<import3.ChartComponent>('chart',viewFactory_ChartComponent_Host0,import3.ChartComponent);
const styles_ChartComponent:any[] = ['[_nghost-%COMP%] {display: block;}'];
var renderType_ChartComponent:import0.RenderComponentType = (null as any);
class _View_ChartComponent0 extends import1.AppView<import3.ChartComponent> {
  _el_0:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement) {
    super(_View_ChartComponent0,renderType_ChartComponent,import6.ViewType.COMPONENT,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import2.AppElement {
    const parentRenderNode:any = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
    this._el_0 = this.renderer.createElement(parentRenderNode,'canvas',(null as any));
    this.init([],[this._el_0],[],[]);
    return (null as any);
  }
}
export function viewFactory_ChartComponent0(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement):import1.AppView<import3.ChartComponent> {
  if ((renderType_ChartComponent === (null as any))) { (renderType_ChartComponent = viewUtils.createRenderComponentType('/Users/sivasakthisoftwares/Desktop/Android/ionicdashboard/node_modules/ng2-chartjs2/components/chart.d.ts class ChartComponent - inline template',0,import9.ViewEncapsulation.Emulated,styles_ChartComponent,{})); }
  return new _View_ChartComponent0(viewUtils,parentInjector,declarationEl);
}
